tfs/src/main/java/hudson/plugins/tfs/rm/ConnectReleaseWebHookEvent.java

```diff
@@ -0,0 +1,319 @@
+            if (webHookItem.getWebHookName().equalsIgnoreCase(webHookName)) {
+                webHook = webHookItem;
+                logger.fine(String.format("WebHook found for webhook name %s", webHookName));
+                break;
+            }
+        }
+
+        if (webHook == null) {
+            throw new InvalidParameterException(String.format("Cannot find webhook with the name %s", webHookName));
+        }
+
+        return webHook;
+    }
+
+    private void createReleaseWebHook(final ReleaseWebHookResource resource) {
+        if (resource == null) {
+            throw new InvalidParameterException("event parameter is null");
+        }
+
+        String payloadUrl = validateAndGetPayloadUrl(resource);
+        String secret = resource.getSecret();
+        String webHookName = resource.getWebHookName();
+
+        if (webHookName == null || webHookName.isEmpty()) {
+            throw new InvalidParameterException("webhook name is empty");
+        }
+
+        List<ReleaseWebHook> releaseWebHooks = ReleaseWebHookHelper.getReleaseWebHookConfigurations();
+        for (ReleaseWebHook webHook : releaseWebHooks) {
+            if (webHook.getPayloadUrl().equalsIgnoreCase(payloadUrl)) {
+                logger.fine(String.format("WebHook found for payloadUrl %s already exists", payloadUrl));
+                return;
+            }
+        }
+
+        ReleaseWebHook webHook = new ReleaseWebHook(webHookName, payloadUrl, secret);
+        releaseWebHooks.add(webHook);
+        ReleaseWebHookHelper.saveReleaseWebHookConfigurations(releaseWebHooks);
+    }
+
+    private void deleteReleaseWebHook(final ReleaseWebHookResource resource) {
+        if (resource == null) {
+            throw new InvalidParameterException("event parameter is null");
+        }
+
+        final TeamPluginGlobalConfig config = TeamPluginGlobalConfig.get();
+        if (config == null) {
+            throw new InternalError("Cannot load TFS global configuration");
+        }
+
+        String webHookNameToDelete = resource.getWebHookName();
+        ReleaseWebHook webHook = validateAndGetReleaseWebHookByName(webHookNameToDelete);
+
+        for (final Item projectItem : Jenkins.getActiveInstance().getAllItems()) {
+            if (projectItem instanceof AbstractProject) {
+                AbstractProject project = (AbstractProject) projectItem;
+
+                ReleaseWebHookAction action = getReleaseWebHookActionFromProject(project);
+                if (action != null) {
+                    for (ReleaseWebHookName webHookName : action.getWebHookNames()) {
+                        if (webHookName.getWebHookName().equalsIgnoreCase(webHookNameToDelete)) {
+                            logger.fine(String.format("WebHook %s is referenced in project %s. Cannot delete until all the references are removed.", webHookNameToDelete, project.getName()));
+                            return;
```

Comment:
throw exception instead of returning without deleting?

Reply:
fixed.